### DISCUSSION TITLE
feature request: Enable "rename..." in context menu even without permission

### DISCUSSION THREAD

**Discussion by arisoda** [linuxmint/discussions#480] [Ideas]
Since Nemo is a UI, it should be easy to copy filenames to your clipboard, and the operation to do so should not matter depending on the file permissions at hand.

I propose to ALWAYS enable the "rename..." option in the context menu of a file, because this allows for quickly COPYING the filename. This is convenient because then it's a CONSISTENT way of copying the filename of ANY file. I'm aware you can click properties and can copy the filename that way, universally, but pressing f2, then ctrl+c, escape, is MUCH QUICKER, and does not need a SLOW properties window to pop up (esp. slow for large files). Also, this properties window automatically selects the FULL filename, **including extension**.

If you **lack permission** to modify filenames, it should only prevent you from doing so if you ACTUALLY change it and **hit enter**. This change should also make the UI feel more fluid and less rigid. Why stop the user in its tracks when he may not actually change the filename?

What do people think? Good idea / bad idea?

**Comment by claudiux** [comment]
This nemo action solves your problem: https://cinnamon-spices.linuxmint.com/actions/view/40

**Comment by mtwebster** [comment]
If you right-click-Copy a file, then attempt to paste it somewhere that is *not* a file manager (or file handler for that specific file type), the path will be pasted instead.

**Reply by arisoda** [reply]
sry for late reply here, I do see your point, but I would really argue this is kinda missing the point, tbh. First of all, the path is something very different than just the name of the file; it would require far more fiddling around (open text editor, remove the rest, etc) to go from path to just file name. It just doesn't work well in work flows. 

And second, most importantly, I don't see any reason for this feature to _not_ be implemented, is there? It'd only streamline the UI _more_, and it's _far_ easier this way to copy the file name... It just makes more sense, even on an intuitive level, as well as for people coming from Windows. There is literally no downside, I'd say. Wouldn't it be nicer to have this, honestly? 

I know it requires more work, but I think it is really worth it. Please consider.

**Reply by mtwebster** [reply]
You're requesting we break the behavior of something in order to make it more convenient for you. We're not going to do this.

You can use an action as mentioned above. In nemo 6.4 I'm hoping to be able to allow actions to have shortcuts assigned to them, which might make this more appealing to you.
